core/src/main/java/cz/xtf/core/bm/NamespaceManager.java

```diff
@@ -0,0 +1,119 @@
+package cz.xtf.core.bm;
+
+import static cz.xtf.core.config.OpenShiftConfig.OPENSHIFT_NAMESPACE;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.engine.descriptor.MethodBasedTestDescriptor;
+import org.junit.platform.engine.TestDescriptor;
+
+import cz.xtf.core.config.OpenShiftConfig;
+import cz.xtf.core.config.XTFConfig;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import io.fabric8.kubernetes.api.builder.Visitor;
+import io.fabric8.kubernetes.api.model.NamespaceBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class NamespaceManager {
+    /**
+     * Used only in case when test cases are executed in different namespaces when:
+     * -Dxtf.openshift.namespace.per.testcase=true
+     *
+     * Maps testcase -> namespace
+     */
+    private static final Map<String, String> testcaseToNamespaceMap = new HashMap<String, String>();
+
+    /**
+     * Used only in case when test cases are executed in different namespaces when:
+     * -Dxtf.openshift.namespace.per.testcase=true
+     *
+     * @return Map testcase -> namespace
+     */
+    private static Map<String, String> getTestcaseToNamespaceMap() {
+        return testcaseToNamespaceMap;
+    }
+
+    /**
+     * Used only in case when test cases are executed in different namespaces when:
+     * -Dxtf.openshift.namespace.per.testcase=true
+     *
+     * @return return namespace for currently running testcase or null if not present
+     */
+    public static String getNamespaceForTestcase(String testcase) {
+        return getTestcaseToNamespaceMap().get(testcase);
+    }
+
+    public static void createIfDoesNotExistsProject(String namespace) {
```

Comment:
`NamespaceManager` puts logic for creating namespace to one place so it can be called not just by `ProjectCreator`. For example it's used in `DockerImageMetadata`: https://github.com/xtf-cz/xtf/pull/490/files#diff-d4e5929b72c1d68c1e28f2ca9827840373ad1cb557aed670eeeefedd0400d7afR84

Reply:
Ok. To be honest I do not understand why DockerImageMetadata should create project, but I don't want to go to that rabbit hole now :) It is your decision and I will accept it. 

Resolving this then.